test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+static const char *command_line5[] = {"parse_opt", "-vd", "-Pomy_store",
+  "-c"
+  "string_opt",
+  "-PT", NULL};
+static const char *command_line6[] = {
+  "parse_opt", "-dv", "-Pomy_store", "-cstring_opt", "-PT", NULL};
+static const char *command_line7[] = {
+  "parse_opt", "-ev", "-cstring_opt", "-Pomy_store", "-PT", "-f", "22", NULL};
+static const char *command_line8[] = {"parse_opt", "-evd", "-Pomy_store", "-PT", "-f22", NULL};
+static const char *command_line9[] = {"parse_opt", "-v", "-Pomy_store", "-PT", NULL};
+
+static const char **command_lines[] = {
+  command_line0,
+  command_line1,
+  command_line2,
+  command_line3,
+  command_line4,
+  command_line5,
+  command_line6,
+  command_line7,
+  command_line8,
+  command_line9,
+};
+
+static TEST_OPTS expected[] = {
+  {NULL, NULL, {0}, NULL, (char *)"builddir", NULL, 0, NULL, NULL, false, false, false, false, 0, 0,
+    21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, (char *)"builddir", NULL, 0, NULL, NULL, false, false, false, false, 0, 0,
+    21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  /* If -PT is used, the tiered_storage source is set to dir_store, even if -Po is not used. */
+  {NULL, NULL, {0}, NULL, NULL, (char *)"dir_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+
+  /* Extended tests */
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+};
+
+static EXTENDED_OPTS x_expected[] = {
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+
+  {(char *)"string_opt", true, false, 0},
+  {(char *)"string_opt", true, false, 0},
+  {(char *)"string_opt", false, true, 22},
+  {NULL, true, true, 22},
+  {NULL, false, false, 0},
+};
+
+/*
+ * report --
+ *     Show any changed fields in the options.
+ */
+static void
+report(const TEST_OPTS *opts, EXTENDED_OPTS *x_opts)
+{
+#define REPORT_INT(o, field) \
+    if (o->field != 0)       \
+    printf(#field ": %" PRIu64 "\n", (uint64_t)o->field)
```

Comment:
Not sure if this is you or clang-format, but both of these macros should have indented the `printf` lines after the `if` line.

Reply:
This was clang-format.  It turns out that if you do the do-while(0) trick, it indents properly, otherwise apparently not.  So I'll use do-while for all of these.